### PR TITLE
Integration ticketservice

### PR DESCRIPTION
**Pull request by fwald**
Added integrations for ticket-service. A ticket is generated when we get a new claim created through user audio.

### CODE REVIEW

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    sb.append("\nClaim audio url: \n");
+    sb.append(event.getAudioURL());
+    sb.append("\nCreated at: \n");
+    sb.append(timestamp);
+
+    String description = sb.toString();
+
+    TicketDto ticket = new TicketDto (
+      event.getUserId(),
+      "claims-service@hedvig.com",
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]:
```suggestion
      "claims-service",
```

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      "Unassigned",
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]: Shouldn't unassigend be represented by null instead

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/dto/TicketDto.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Value;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+
+
+@Value
+public class TicketDto {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/dto/TicketDto.java]: make sure this is up to date

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  private final TicketService ticketService;
+
+  @Autowired
+  public TicketEventListener ( TicketService ticketService ) {
+    this.ticketService = ticketService;
+  }
+
+  @EventHandler
+  public void on(ClaimCreatedEvent event, @Timestamp Instant timestamp) {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]: would be cool if you could listen to more events and the ticket would update itself accordingly.

Comment (fwald) [reply src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]: Yes, that would be great. I think that we should merge this to begin with for testing purposes though and then add more features like that later on

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java (hunk 2 above)
Comment (fwald) [reply src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]: Right now "Unassigned" is used, can of course change this if necessary

File: src/main/java/com/hedvig/claims/serviceIntegration/paymentService/PaymentServiceImpl.java [hunk 5]
```diff
@@ -4,6 +4,8 @@
 import com.hedvig.claims.serviceIntegration.paymentService.dto.PayoutRequest;
 import com.hedvig.claims.serviceIntegration.paymentService.dto.TransactionStatus;
 import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/serviceIntegration/paymentService/PaymentServiceImpl.java]: You dont need dat :)
```suggestion
```

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    StringBuilder sb = new StringBuilder();
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketEventListener.java]: I think its better if we send the raw data(ex. memberId, claimId, URL etc.) that we need to create a ticket to Ticket-Service, and construct the Ticket *inside* ticker-service. The same goes to line 37, that we create the ticket object. 

Claims service should *not* care/know what means a ticket, which types the ticket has and so on.

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketServiceClient.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+
+@FeignClient(value = "ticket-service", url = "${hedvig.ticket-service.url:ticket-service}")
+
+public interface TicketServiceClient {
+
+  @PostMapping( value="/_/tickets/new/")
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketServiceClient.java]:
```suggestion
  @PostMapping( value="/_/ticket/create")
```

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketStatus.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.hedvig.claims.serviceIntegration.ticketService;
+
+public enum TicketStatus {
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketStatus.java]: Do we actually want to have this here?

File: src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketType.java [hunk 9]
```diff
@@ -0,0 +1,9 @@
+package com.hedvig.claims.serviceIntegration.ticketService;
+
+public enum TicketType {
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/serviceIntegration/ticketService/TicketType.java]: do we care about the type inside claims-service? 😄

**Pull request merged**